### PR TITLE
Download.sh that autodetects current release from github

### PR DESCRIPTION
**Pull request by DarwinJS**
This version detects the currently available version on github.  
Details:
- It uses sed to parse json - so hopefully can be deployed universally without any runtime installs
- self-documenting comment on how to pull directly from the web
- only tested on ubuntu and centos - could use testing on other supported platforms.
- the sed string might be able to be cleaned up with someone who knows sed better than I (like everyone)

### CODE REVIEW

File: tools/download.sh [hunk 1]
```diff
@@ -25,21 +33,22 @@ case "$OSTYPE" in
...
             ubuntu)
                 if ! hash curl 2>/dev/null; then
                     echo "curl not found, installing..."
                     sudo apt-get install -y curl
                 fi
+                get_currentpshversion
 
                 case "$VERSION_ID" in
                     14.04)
-                        package=powershell_6.0.0-alpha.18-1ubuntu1.14.04.1_amd64.deb
```
Comment (TravisEz13) [inline tools/download.sh]: Where installation via package manager is available, can we switch to using that?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
The build is failing.  It should be fixed

File: tools/download.sh (hunk 1 above)
Comment (DarwinJS) [inline tools/download.sh]: We're on the same page!  I am thinking that when I add the repos I should make the initial install request *specifying* the version as the version that comes down in the version tag on git.  That way if there happens to a sync issue where a newer than the latest "official" release is hanging around, this code only installs the official release tagged as latest on git?

**Comment by DarwinJS** [comment]
Another question - is the "Centos" release really only good for Centos - or also Rhel?  Right now the logic limits it to Centos.

File: tools/download.sh (hunk 1 above)
Comment (TravisEz13) [inline tools/download.sh]: I think it should just install the latest version from the repo.  Eventually, we will have a non-pre-release version, and I think then we have a decision to make, but currently, if it's in the repo, it's ready.

Comment (DarwinJS) [inline tools/download.sh]: Suits me - easier to code ;)

**Comment by TravisEz13** [comment]
CentOS, my understanding is that it should work on the same version of Rhel.

**Comment by DarwinJS** [comment]
I've made a proposal here: https://github.com/PowerShell/PowerShell/issues/3547

**Comment by DarwinJS** [comment]
@TravisEz13 - is every .sh that is uploaded tested against every distro - or is there a way to specify which ones run on which test targets?

**Comment by TravisEz13** [comment]
Which .sh file are we talking about?  I doubt there is a way to specify which distro it is tested on.  that would have to be written.

**Comment by TravisEz13** [comment]
What do you want to do with this PR?

**Comment by DarwinJS** [comment]
I would like to merge it, however, I don't have a way to test the OSX code (that now uses brew.)

Would be especially helpful to test on a pristine system.

**Comment by TravisEz13** [comment]
If you fix the merge conflicts, I can probably help you test it.

**Comment by DarwinJS** [comment]
@TravisEz13 - this is superceded by #3608

**Comment by TravisEz13** [comment]
Closing because this is superseded

**Pull request closed without merging**